include/TGUI/Widgets/MessageBox.hpp

```diff
@@ -313,6 +333,12 @@ TGUI_MODULE_EXPORT namespace tgui
         void connectButtonPressSignal(std::size_t buttonIndex);
 
 
+        /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        /// Adds a button to the MessageBox.
+        /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        void addButtonImpl(const String& caption);
```

Comment:
I separated `addButton()`'s code in this way to prevent rearranging the `MessageBox` more than necessary when using `changeButtons()` to add multiple buttons.